### PR TITLE
Fixed fd leak on server_dyn_res alarm

### PR DESCRIPTION
**Pull request by bhroam**
#### Describe Bug or Feature
When server_dyn_res times out, the scheduler leaks 2 fd (pipes).

#### Describe Your Change
When the server_dyn_res times out, close the pipes.

#### Attach Test and Valgrind Logs/Output
Here's lsof output after server_dyn_res timed out twice:
pbs_sched 65501 root    7u     IPv4             106216      0t0       TCP henke:37104->henke:onep-tls (ESTABLISHED)
pbs_sched 65501 root    8u     IPv4             107062      0t0       TCP henke:37106->henke:onep-tls (ESTABLISHED)
pbs_sched 65501 root    9u  a_inode               0,14        0     14276 [eventpoll]
pbs_sched 65501 root   10r     FIFO               0,13      0t0    104420 pipe
pbs_sched 65501 root   11w     FIFO               0,13      0t0    104420 pipe
pbs_sched 65501 root   12r     FIFO               0,13      0t0    122584 pipe
pbs_sched 65501 root   13w     FIFO               0,13      0t0    122584 pipe

Here's after the fix:
pbs_sched 103751 root    7u     IPv4             150318      0t0       TCP henke:37136->henke:onep-tls (ESTABLISHED)
pbs_sched 103751 root    8u     IPv4             150320      0t0       TCP henke:37138->henke:onep-tls (ESTABLISHED)
pbs_sched 103751 root    9u  a_inode               0,14        0     14276 [eventpoll]

### CODE REVIEW

**Review by arungrover — APPROVED** [review]
LGTM

File: src/scheduler/server_info.cpp [hunk 1]
```diff
@@ -776,18 +776,21 @@ query_server_dyn_res(server_info *sinfo)
...
+							pipe_err = errno;
+						} else
+							k = strlen(buf);
+						if (fp != NULL)
+							fclose(fp);
 					} else
-						k = strlen(buf);
-					if (fp != NULL)
-						fclose(fp);
+						close(pdes[0]);
```
Comment (vstumpf) [inline src/scheduler/server_info.cpp:793]: should pdes[0] be closed all the time? not just when ret is <= 0?

Comment (bhroam) [reply src/scheduler/server_info.cpp:793]: pdes[0] is fp, so when you fclose(fp), it should close pdes[0].

Comment (vstumpf) [reply src/scheduler/server_info.cpp:793]: Duh, of course.

**Review by vstumpf — APPROVED** [review]
LGTM

**Pull request merged**